cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -498,46 +498,75 @@ public void programRunRecord(HttpRequest request, HttpResponder responder,
    * Get program runtime args.
    */
   @GET
-  @Path("/apps/{app-id}/{program-type}/{program-id}/runtimeargs")
+  @Path("/apps/{app-name}/{program-type}/{program-name}/runtimeargs")
   public void getProgramRuntimeArgs(HttpRequest request, HttpResponder responder,
                                     @PathParam("namespace-id") String namespaceId,
-                                    @PathParam("app-id") String appId,
+                                    @PathParam("app-name") String appName,
                                     @PathParam("program-type") String type,
-                                    @PathParam("program-id") String programId) throws BadRequestException,
+                                    @PathParam("program-name") String programName) throws BadRequestException,
+    NotImplementedException, NotFoundException {
+    getProgramRuntimeArgs(request, responder, namespaceId, appName, ApplicationId.DEFAULT_VERSION, type, programName);
+  }
+
+  /**
+   * Get program runtime args.
+   */
+  @GET
+  @Path("/apps/{app-name}/versions/{app-version}/{program-type}/{program-name}/runtimeargs")
```

Comment:
didn't mean for it to be removed. It seems like a useful API, but my question is what is the plan for the old APIs? Will they continue to exist for ever or are they just for backward compatibility. If it is just for backward compatibility, then those APIs should be deprecated.